File: 2527-count-subarrays-with-fixed-bounds/count-subarrays-with-fixed-bounds.java
class Solution {
    public long countSubarrays(int[] nums, int minK, int maxK) {
        int n = nums.length;
        long ans = 0;
        int minPos = -1, maxPos = -1, badPos = -1;
        for (int i = 0; i < n; i++) {
            if (nums[i] < minK || nums[i] > maxK) badPos = i;
            if (nums[i] == minK) minPos = i;
            if (nums[i] == maxK) maxPos = i;
            ans += Math.max(0, Math.min(minPos, maxPos) - badPos);
        }
        return ans;
    }
}

File: 2527-count-subarrays-with-fixed-bounds/count-subarrays-with-fixed-bounds.py
class Solution:
    def countSubarrays(self, nums, minK, maxK):
        result = 0
        lastMin = -1
        lastMax = -1
        lastInvalid = -1
        
        for i in range(len(nums)):
            if nums[i] < minK or nums[i] > maxK:
                lastInvalid = i
            if nums[i] == minK:
                lastMin = i
            if nums[i] == maxK:
                lastMax = i
            
            if lastMin != -1 and lastMax != -1:
                result += max(0, min(lastMin, lastMax) - lastInvalid)
        
        return result
